Python/compile.c

```diff
@@ -2695,7 +2705,11 @@ compiler_lambda(struct compiler *c, expr_ty e)
     if (co == NULL)
         return 0;
 
-    compiler_make_closure(c, co, funcflags, qualname);
+    if (!compiler_make_closure(c, co, funcflags, qualname)) {
```

Comment:
```
    if (co == NULL)
        return 0;
```
Just to make sure, shouldn't the statement above also `Py_DECREF(qualname)`?

Reply:
Oh, there is a reference leak in the co == NULL test above. But it's unrelated to my PR. I suggest to fix it separately.

Reply:
Opened https://github.com/python/cpython/pull/24382